cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java

```diff
@@ -137,14 +143,13 @@ public final String toString() {
 
   @Override
   public boolean equals(Object o) {
-    if (this == o) {
-      return true;
-    }
-    if (o == null || getClass() != o.getClass()) {
-      return false;
-    }
+    if (this == o) return true;
+    if (!(o instanceof EntityId)) return false;
+
     EntityId entityId = (EntityId) o;
-    return Objects.equals(entity, entityId.entity);
+
+    return getEntity() == entityId.getEntity();
```

Comment:
this is strange. so we return true if the entity type is equal but everything else is different? Shouldn't it be something like:

```
if (getEntityType() != entityId.getEntityType()) {
  return false;
}

return Objects.equals(this, entityId);
```

Although it seems like the original code also did it the same way